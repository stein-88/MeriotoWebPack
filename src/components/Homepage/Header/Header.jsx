import React from 'react'
import schema from './data'
import s from './Header.scss'

const Header = () => {
    const { botao, titulo } = schema
    const { link, text } = botao
    return (
        <header>
            <section className={`${s.bgblack} w-100 py-4 mb-5`}>
                <section className={s.imgheader} />
                <div className="container">
                    <div className="row">
                        <div className="col-12 text-center text-white">
                            {titulo && <h2 className="py-4 h1">{titulo}</h2>}
                            {link && text && (
                                <a className={s.botaoreg} href={link}>
                                    <strong>{text}</strong>
                                </a>
                            )}
                        </div>
                    </div>
                </div>
            </section>
        </header>
    )
}

export default Header
