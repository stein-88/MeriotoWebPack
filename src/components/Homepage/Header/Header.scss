@import "../../../stylesheets/global.scss";

.bgblack {
  background-color: black;
  min-height: calc(100vh - 56px);

  .imgheader {
    min-height: 450px;
    background-image: url("../../../statics/images/homehead1.png");
    background-position: center center;
    background-size: cover;
    background-repeat: no-repeat;
    position: relative;
    max-width: 1380px;
    margin: 0 auto;
  }

  .botaoreg {
    cursor: pointer;
    display: inline-block;
    padding: 10px 20px;
    width: 100%;
    max-width: 280px;
    color: #424141;
    text-align: center;
    outline: 0;
    background-color: $coramarela;
    border: 2px solid $coramarela;
    border-radius: 15px;
    font-size: 20px;

    &:hover {
      background-color: darken($color: $coramarela, $amount: 12);
    }
  }
}
