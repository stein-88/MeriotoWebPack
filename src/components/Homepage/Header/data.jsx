const schema = {
    titulo: 'Join the Reuse Revolution for Free items!',
    botao: {
        link: 'https://clavo.globechain.com/register',
        text: 'Register',
    },
}

export default schema
