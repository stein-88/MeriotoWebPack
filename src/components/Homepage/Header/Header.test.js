import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'

describe('testing Header', () => {
    const compHeader = renderer.create(<Header />)
    test('Should render without props', () => {
        const jsonHeader = compHeader.toJSON()
        expect(jsonHeader).toMatchSnapshot()
    })
    test('Should have 1 img, 1 title and 1 button', () => {
        const resultIMG = compHeader.root.findAllByProps({ className: 'imgheader' })
        const resultTitle = compHeader.root.findAllByProps({ className: 'py-4 h1' })
        const resultBtn = compHeader.root.findAllByProps({ className: 'botaoreg' })
        expect(resultIMG).toHaveLength(1)
        expect(resultTitle).toHaveLength(1)
        expect(resultBtn).toHaveLength(1)
    })
})