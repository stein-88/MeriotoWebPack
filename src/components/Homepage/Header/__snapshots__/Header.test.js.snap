// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header Should render without props 1`] = `
<header>
  <section
    className="bgblack w-100 py-4 mb-5"
  >
    <section
      className="imgheader"
    />
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 text-center text-white"
        >
          <h2
            className="py-4 h1"
          >
            Join the Reuse Revolution for Free items!
          </h2>
          <a
            className="botaoreg"
            href="https://clavo.globechain.com/register"
          >
            <strong>
              Register
            </strong>
          </a>
        </div>
      </div>
    </div>
  </section>
</header>
`;
